### ISSUE TITLE
Assert in forced align layer

### ISSUE THREAD

**Issue by Atticus1806** [rwth-i6/returnn#1086]
The `forced_align` layer includes the following assert: 
`assert align_target.output.shape == (None,) and align_target.output.dim == logits_data.dim - 1` , but I am wondering why this makes sense. Does that mean that the blank symbol implicitly is not suppposed to be part of the vocab? 
Before the changes to the blank_idx in that layer it was always, ` logits_data.dim - 1` so that afterwards the dims implicitly would match, but I am thinking that this might be very wrong if `blank_idx` is set by the user now, then I think the check would have to be `align_target.output.dim == logits_data.dim` to be correct, even for the case that the blank symbol is already in the align_target vocab.
